aptos-move/aptos-vm/src/data_cache.rs

```diff
@@ -235,3 +233,30 @@ impl<S: StateView> AsMoveResolver<S> for S {
         StorageAdapter::new(self)
     }
 }
+
+impl<'a, S: StateView> StateValueMetadataResolver for StorageAdapter<'a, S> {
+    fn get_state_value_metadata(
+        &self,
+        state_key: &StateKey,
+    ) -> anyhow::Result<Option<Option<StateValueMetadata>>> {
+        let maybe_state_value = self.state_store.get_state_value(state_key)?;
+        Ok(maybe_state_value.map(|state_value| state_value.into_metadata()))
+    }
+}
+
+// We need to implement StateView for adapter because:
+//   1. When processing write set payload, storage is accessed
+//      directly.
+//   3. When stacking Storage adapters on top of each other, e.g.
+//      in epilogue.
+impl<'a, S: StateView> TStateView for StorageAdapter<'a, S> {
+    type Key = StateKey;
+
+    fn get_state_value(&self, state_key: &Self::Key) -> anyhow::Result<Option<StateValue>> {
+        self.state_store.get_state_value(state_key)
+    }
+
+    fn get_usage(&self) -> anyhow::Result<StateStorageUsage> {
+        self.state_store.get_usage()
```

Comment:
What is usage here?

Reply:
I think this is needed for Move functions to query how many bytes are used in storage, etc. for pricing.